modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -1999,7 +1998,7 @@ private void waitPartitionRelease(
         releaseLatch.countDown();
 
         // For compatibility with old version where joining nodes are not waiting for latch.
-        if (localJoinExchange() && !cctx.exchange().latch().canSkipJoiningNodes(initialVersion()))
```

Comment:
Looks like this clause was always `false` (canSkipJoiningNodes always return `true`). 
So in PR you can remove it completely